repository/src/main/java/org/alfresco/repo/action/executer/ScriptActionExecuter.java

```diff
@@ -152,9 +152,8 @@ protected void executeImpl(Action action, NodeRef actionedUponNodeRef)
             {
                 // get the references we need to build the default scripting data-model
                 String userName = this.serviceRegistry.getAuthenticationService().getCurrentUserName();
-                NodeRef personRef = this.personService.getPerson(userName);
-                NodeRef homeSpaceRef = (NodeRef)nodeService.getProperty(personRef, ContentModel.PROP_HOMEFOLDER);
-                
+                NodeRef personRef = this.personService.getPersonOrNull(userName);

```

Comment:
Please add comments to explain these two line changes.  We need something to the affect of when a script is executed as a system user there is no person to home space to provide to the scripting framework.

Reply:
Added comments above the two lines. Similar Changes haven been already in-place for SAP Connector.
https://github.com/Alfresco/sap-content-connector-repo/blob/36bcd4efeecba7b7a529af49f6706f22aa8851fa/src/main/java/com/alfresco/integrations/sap/repo/action/PatchedScriptActionExecuter.java